assets/javascripts/discourse/services/chat.js

```diff
@@ -545,6 +548,27 @@ export default Service.extend({
     });
   },
 
+  _subscribeToChannelStatusChange() {
+    this.messageBus.subscribe("/chat/channel-status", (busData) => {
+      let activeChannel = this.getActiveChannel();
+      if (
+        busData.chat_channel_id === activeChannel?.id &&
+        (busData.status === CHANNEL_STATUSES.readOnly ||
+          busData.status === CHANNEL_STATUSES.archived)
+      ) {
+        // This is not very elegant. Ideally at some point we want to have
+        // a nice reactive magical transformation of the channel status
+        // before the user's very eyes...but for now let's just reload so
+        // they can see they are no longer allowed to chat.
+        window.location.reload();
```

Comment:
I understand your concern: Do you think you could log this somewhere, or inform @markvanlan about this so he could add it in a list of improvements to be made?

Reply:
I did mention it in the internal archive topic, however I will make a new one for this specific todo :)